File: scratch/update_css.js
const fs = require('fs');
const path = require('path');

const files = [
    "app/(storefront)/shop/[slug]/pdp.module.css",
    "components/pdp/ProductInfo.module.css",
    "components/pdp/ProductGallery.module.css",
    "components/pdp/VariantSelector.module.css",
    "components/pdp/QuantitySelector.module.css",
    "components/pdp/AddToCartBar.module.css",
    "components/pdp/ProductTabs.module.css",
    "components/pdp/RelatedProducts.module.css",
    "components/cart/CartItem.module.css",
    "components/cart/CartSummary.module.css",
    "components/cart/EmptyCart.module.css",
    "components/checkout/CheckoutForm.module.css",
    "components/checkout/CheckoutSummary.module.css",
    "components/account/AccountShell.module.css",
    "components/account/AccountNav.module.css",
    "components/account/DashboardClient.module.css",
    "components/account/OrderCard.module.css",
    "components/account/ProfileForm.module.css",
];

const baseDir = "c:\\Users\\rajni\\OneDrive\\Desktop\\All Projects\\RFC STORE";

function replaceTokens(content) {
    let newContent = content;
    const replacements = {
        "var(--color-primary)": "var(--rfc-text)",
        "var(--color-secondary)": "var(--rfc-accent)",
        "var(--color-secondary-dark)": "var(--rfc-accent-dark)",
        "var(--color-on-primary)": "var(--rfc-text-inv)",
        "var(--color-on-secondary)": "var(--rfc-text-inv)",
        "var(--color-background)": "var(--rfc-bg)",
        "var(--color-surface-container-lowest)": "var(--rfc-surface)",
        "var(--color-surface-container-low)": "var(--rfc-bg)",
        "var(--color-surface-container)": "var(--rfc-border)",
        "var(--color-outline-variant)": "var(--rfc-border)",
        "var(--color-outline)": "var(--rfc-text-subtle)",
        "var(--color-error)": "var(--rfc-error)",
        "var(--color-error-container)": "var(--rfc-error-bg)",
        "var(--color-on-error-container)": "var(--rfc-error)",
        "var(--color-on-surface-variant)": "var(--rfc-text-muted)",
        "var(--color-on-surface)": "var(--rfc-text)",
        "var(--radius-default)": "var(--radius-md)",
        "rgba(11, 28, 48, 0.45)": "var(--rfc-text-subtle)",
        "rgba(11, 28, 48, 0.25)": "var(--rfc-border)",
        "rgba(11, 28, 48, 0.35)": "var(--rfc-text-subtle)",
        "rgba(11, 28, 48, 0.2)": "var(--rfc-border)",
        "rgba(11, 28, 48, 0.5)": "var(--rfc-text-muted)",
        "rgba(11, 28, 48, 0.65)": "var(--rfc-text-muted)",
        "rgba(11, 28, 48, 0.55)": "var(--rfc-text-muted)",
        "rgba(11, 28, 48, 0.4)": "var(--rfc-text-subtle)",
        "rgba(11, 28, 48, 0.3)": "var(--rfc-text-subtle)",
        "rgba(11, 28, 48, 0.6)": "var(--rfc-text-muted)",
        "rgba(11, 28, 48, 0.7)": "var(--rfc-text-muted)",
        "rgba(11, 28, 48, 0.12)": "var(--rfc-border)",
        "#0b1c30": "var(--rfc-dark)",
        "#1a2a3a": "var(--rfc-dark-surface)",
        "#ffffff": "var(--rfc-surface)",
        "#fff": "var(--rfc-surface)",
        "#0a0e14": "var(--rfc-dark)",
        "#1a2030": "var(--rfc-dark-surface)"
    };
    
    for (const [key, value] of Object.entries(replacements)) {
        newContent = newContent.split(key).join(value);
    }
    return newContent;
}

files.forEach(filePath => {
    const fullPath = path.join(baseDir, filePath);
    if (!fs.existsSync(fullPath)) {
        console.log(`File not found: ${fullPath}`);
        return;
    }
    let content = fs.readFileSync(fullPath, "utf-8");
    content = replaceTokens(content);

    if (filePath.includes("ProductInfo.module.css")) {
        content = content.split("color: var(--rfc-accent);").join("color: var(--rfc-text);");
    }

    if (filePath.includes("VariantSelector.module.css")) {
        content = content.split("background-color: var(--rfc-text);").join("background-color: var(--rfc-accent-light);");
        content = content.split("color: var(--rfc-text-inv);").join("color: var(--rfc-accent);");
        content = content.split("border-color: var(--rfc-text);").join("border-color: var(--rfc-accent);");
    }

    if (filePath.includes("AddToCartBar.module.css")) {
        content = content.split(".wishlistBtn {\n  display: flex;").join(".wishlistBtn {\n  background-color: var(--rfc-dark);\n  color: var(--rfc-surface);\n  display: flex;");
        content = content.split("color: var(--rfc-text-muted);\n  cursor: pointer;").join("color: var(--rfc-text-inv-muted);\n  cursor: pointer;");
    }

    if (filePath.includes("CartItem.module.css")) {
        content = content.split("background-color: var(--rfc-error-bg);").join("background-color: transparent;");
    }

    if (filePath.includes("CheckoutForm.module.css")) {
        content = content.split("background-color: #fffbeb;").join("background-color: var(--rfc-warning-bg);");
        content = content.split("border: 1px solid #f59e0b;").join("border: 1px solid var(--rfc-warning);");
        content = content.split("color: #92400e;").join("color: var(--rfc-warning);");
        content = content.split("background-color: var(--rfc-text);").join("background-color: var(--rfc-accent);");
        content = content.split("background-color: var(--rfc-dark-surface);").join("background-color: var(--rfc-accent-dark);");
        content = content.split("box-shadow: 0 0 0 3px rgba(10, 14, 20, 0.08);").join("box-shadow: 0 0 0 3px rgba(230,57,70,0.1);");
        content = content.split("box-shadow: 0 0 0 3px rgba(10,14,20,0.08);").join("box-shadow: 0 0 0 3px rgba(230,57,70,0.1);");
    }

    if (filePath.includes("AccountNav.module.css")) {
        content = content.split("background: var(--rfc-accent);").join("background: transparent;\n  color: var(--rfc-accent);\n  border-left: 3px solid var(--rfc-accent);");
        content = content.split("color: var(--rfc-text-inv);").join("color: var(--rfc-accent);");
        content = content.split("background: var(--rfc-accent-dark);").join("background: var(--rfc-bg);");
    }
    
    if (filePath.includes("ProfileForm.module.css")) {
        content = content.split("box-shadow: 0 0 0 3px rgba(10,14,20,0.08);").join("box-shadow: 0 0 0 3px rgba(230,57,70,0.1); border-color: var(--rfc-accent);");
    }

    fs.writeFileSync(fullPath, content, "utf-8");
    console.log(`Updated ${filePath}`);
});

console.log("Done");

File: scratch/update_css.py
import os
import re

files = [
    r"app\(storefront)\shop\[slug]\pdp.module.css",
    r"components\pdp\ProductInfo.module.css",
    r"components\pdp\ProductGallery.module.css",
    r"components\pdp\VariantSelector.module.css",
    r"components\pdp\QuantitySelector.module.css",
    r"components\pdp\AddToCartBar.module.css",
    r"components\pdp\ProductTabs.module.css",
    r"components\pdp\RelatedProducts.module.css",
    r"components\cart\CartItem.module.css",
    r"components\cart\CartSummary.module.css",
    r"components\cart\EmptyCart.module.css",
    r"components\checkout\CheckoutForm.module.css",
    r"components\checkout\CheckoutSummary.module.css",
    r"components\account\AccountShell.module.css",
    r"components\account\AccountNav.module.css",
    r"components\account\DashboardClient.module.css",
    r"components\account\OrderCard.module.css",
    r"components\account\ProfileForm.module.css",
]

base_dir = r"c:\Users\rajni\OneDrive\Desktop\All Projects\RFC STORE"

def replace_tokens(content):
    # Global replaces
    content = content.replace("var(--color-primary)", "var(--rfc-text)")
    content = content.replace("var(--color-secondary)", "var(--rfc-accent)")
    content = content.replace("var(--color-secondary-dark)", "var(--rfc-accent-dark)")
    content = content.replace("var(--color-on-primary)", "var(--rfc-text-inv)")
    content = content.replace("var(--color-on-secondary)", "var(--rfc-text-inv)")
    content = content.replace("var(--color-background)", "var(--rfc-bg)")
    content = content.replace("var(--color-surface-container-lowest)", "var(--rfc-surface)")
    content = content.replace("var(--color-surface-container-low)", "var(--rfc-bg)")
    content = content.replace("var(--color-surface-container)", "var(--rfc-border)")
    content = content.replace("var(--color-outline-variant)", "var(--rfc-border)")
    content = content.replace("var(--color-outline)", "var(--rfc-text-subtle)")
    content = content.replace("var(--color-error)", "var(--rfc-error)")
    content = content.replace("var(--color-error-container)", "var(--rfc-error-bg)")
    content = content.replace("var(--color-on-error-container)", "var(--rfc-error)")
    content = content.replace("var(--color-on-surface-variant)", "var(--rfc-text-muted)")
    content = content.replace("var(--color-on-surface)", "var(--rfc-text)")
    content = content.replace("var(--radius-default)", "var(--radius-md)")
    content = content.replace("rgba(11, 28, 48, 0.45)", "var(--rfc-text-subtle)")
    content = content.replace("rgba(11, 28, 48, 0.25)", "var(--rfc-border)")
    content = content.replace("rgba(11, 28, 48, 0.35)", "var(--rfc-text-subtle)")
    content = content.replace("rgba(11, 28, 48, 0.2)", "var(--rfc-border)")
    content = content.replace("rgba(11, 28, 48, 0.5)", "var(--rfc-text-muted)")
    content = content.replace("rgba(11, 28, 48, 0.65)", "var(--rfc-text-muted)")
    content = content.replace("rgba(11, 28, 48, 0.55)", "var(--rfc-text-muted)")
    content = content.replace("rgba(11, 28, 48, 0.4)", "var(--rfc-text-subtle)")
    content = content.replace("rgba(11, 28, 48, 0.3)", "var(--rfc-text-subtle)")
    content = content.replace("rgba(11, 28, 48, 0.6)", "var(--rfc-text-muted)")
    content = content.replace("rgba(11, 28, 48, 0.7)", "var(--rfc-text-muted)")
    content = content.replace("rgba(11, 28, 48, 0.12)", "var(--rfc-border)")
    content = content.replace("#0b1c30", "var(--rfc-dark)")
    content = content.replace("#1a2a3a", "var(--rfc-dark-surface)")
    content = content.replace("#ffffff", "var(--rfc-surface)")
    content = content.replace("#fff", "var(--rfc-surface)")
    content = content.replace("#0a0e14", "var(--rfc-dark)")
    content = content.replace("#1a2030", "var(--rfc-dark-surface)")
    return content

for file_path in files:
    full_path = os.path.join(base_dir, file_path)
    if not os.path.exists(full_path):
        print(f"File not found: {full_path}")
        continue
    with open(full_path, "r", encoding="utf-8") as f:
        content = f.read()

    original_content = content
    content = replace_tokens(content)

    # Specific file overrides
    if "ProductInfo.module.css" in file_path:
        content = content.replace("color: var(--rfc-accent);", "color: var(--rfc-text);") # Price should be text not accent
        # Only discount badges get accent, which they already do if mapped correctly

    if "VariantSelector.module.css" in file_path:
        # Selected variant: border var(--rfc-accent), bg var(--rfc-accent-light)
        content = content.replace("background-color: var(--rfc-text);", "background-color: var(--rfc-accent-light);")
        content = content.replace("color: var(--rfc-text-inv);", "color: var(--rfc-accent);")
        content = content.replace("border-color: var(--rfc-text);", "border-color: var(--rfc-accent);")
        # Fix hover
        content = content.replace("border-color: var(--rfc-text);", "border-color: var(--rfc-accent);")

    if "AddToCartBar.module.css" in file_path:
        # Wishlist button: Secondary button: var(--rfc-dark) bg
        content = content.replace(".wishlistBtn {\n  display: flex;", ".wishlistBtn {\n  background-color: var(--rfc-dark);\n  color: var(--rfc-surface);\n  display: flex;")
        content = content.replace("color: var(--rfc-text-muted);\n  cursor: pointer;", "color: var(--rfc-text-inv-muted);\n  cursor: pointer;")
        content = content.replace("border-color: var(--rfc-accent);\n  color: var(--rfc-accent);", "border-color: var(--rfc-accent);\n  color: var(--rfc-accent);")

    if "ProductTabs.module.css" in file_path:
        # Active tab underline: var(--rfc-accent)
        pass # should be handled by --color-secondary replacement

    if "CartItem.module.css" in file_path:
        # Remove button should use var(--rfc-error) color only (not bg)
        content = content.replace("background-color: var(--rfc-error-bg);", "background-color: transparent;")

    if "CheckoutForm.module.css" in file_path:
        # Payment Notice
        content = content.replace("background-color: #fffbeb;", "background-color: var(--rfc-warning-bg);")
        content = content.replace("border: 1px solid #f59e0b;", "border: 1px solid var(--rfc-warning);")
        content = content.replace("color: #92400e;", "color: var(--rfc-warning);")
        # Place Order Btn bg
        content = content.replace("background-color: var(--rfc-text);", "background-color: var(--rfc-accent);")
        content = content.replace("background-color: var(--rfc-dark-surface);", "background-color: var(--rfc-accent-dark);")
        content = content.replace("box-shadow: 0 0 0 3px rgba(10, 14, 20, 0.08);", "box-shadow: 0 0 0 3px rgba(230,57,70,0.1);")
        content = content.replace("box-shadow: 0 0 0 3px rgba(10,14,20,0.08);", "box-shadow: 0 0 0 3px rgba(230,57,70,0.1);")

    if "AccountNav.module.css" in file_path:
        # Active item: var(--rfc-accent) color and left border
        content = content.replace("background: var(--rfc-accent);", "background: transparent;\n  color: var(--rfc-accent);\n  border-left: 3px solid var(--rfc-accent);")
        content = content.replace("color: var(--rfc-text-inv);", "color: var(--rfc-accent);")
        content = content.replace("background: var(--rfc-accent-dark);", "background: var(--rfc-bg);")

    with open(full_path, "w", encoding="utf-8") as f:
        f.write(content)
    print(f"Updated {file_path}")

print("Done")
